src/main/java/org/folio/dew/service/UserPermissionsService.java

```diff
@@ -0,0 +1,49 @@
+package org.folio.dew.service;
+
+import lombok.RequiredArgsConstructor;
+import lombok.Setter;
+import lombok.extern.log4j.Log4j2;
+import org.folio.dew.client.EurekaUserPermissionsClient;
+import org.folio.dew.client.OkapiUserPermissionsClient;
+import org.folio.spring.FolioExecutionContext;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+
+
+@RequiredArgsConstructor
+@Log4j2
+@Service
+public class UserPermissionsService {
+
+  public static final String BULK_EDIT_INVENTORY_VIEW_PERMISSION = "bulk-operations.item.inventory.get";
+  public static final String BULK_EDIT_USERS_VIEW_PERMISSION = "bulk-operations.item.users.get";
+
+  public static final String USER_ITEM_GET_PERMISSION = "users.item.get";
+  public static final String INVENTORY_ITEMS_ITEM_GET_PERMISSION = "inventory.items.item.get";
+  public static final String INVENTORY_STORAGE_HOLDINGS_ITEM_GET_PERMISSION = "inventory-storage.holdings.item.get";
+  public static final String INVENTORY_INSTANCES_ITEM_GET_PERMISSION = "inventory.instances.item.get";
+
+  @Setter
+  @Value("${application.eureka.permissions-model}")
+  private boolean isEurekaPermissionsModel;
+
+  private final FolioExecutionContext folioExecutionContext;
+  private final OkapiUserPermissionsClient okapiUserPermissionsClient;
+  private final EurekaUserPermissionsClient eurekaUserPermissionsClient;
+
+  public List<String> getPermissions() {
+    if (isEurekaPermissionsModel) {
+      var desiredPermissions = getDesiredPermissions();
+      return eurekaUserPermissionsClient.getPermissions(folioExecutionContext.getUserId().toString(),
```

Comment:
Factory for getting permissions client can be used here: FeignClient permissionsClient = getPermissionsClient(String platform).

Reply:
Hi @khandramai , the arguments of clients and return value is different, from my point of view it is possible to use as is.